src/rules/trailingCommaRule.ts

```diff
@@ -20,11 +20,98 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+type OptionValue = "always" | "never" | "ignore";
+
+interface CustomOptionValue {
+    arrays?: OptionValue;
+    exports?: OptionValue;
+    functions?: OptionValue;
+    imports?: OptionValue;
+    objects?: OptionValue;
+    typeLiterals?: OptionValue;
+}
+
 interface Options {
-    multiline?: "always" | "never";
-    singleline?: "always" | "never";
+    multiline: CustomOptionValue;
+    singleline: CustomOptionValue;
 }
 
+const defaultOptions: CustomOptionValue = {
+    arrays: "ignore",
+    exports: "ignore",
+    functions: "ignore",
+    imports: "ignore",
+    objects: "ignore",
+    typeLiterals: "ignore",
+};
+
+const normalizeOptions = (options: Options): Options => {
+    if (typeof options.multiline === "string") {
```

Comment:
consider making this a separate function as both `multiline` and `singleline` are handled the same.